src/queue-manager.js

```diff
@@ -57,62 +115,30 @@ class QueueManager {
     }
 
     if (config.deadHandler) {
+      assert(config.deadQueueUrl, 'When using a dead letter queue, must provide url');
       this.deadHandler = config.deadHandler;
+      this.deadQueueUrl = config.deadQueueUrl;
       this.deadBatchSize = config.deadBatchSize || config.batchSize;
     }
 
-    this.maxReceiveCount = config.maxReceiveCount || 5;
-
-    this.deadQueueName = config.deadQueueName || this.queueName + '_dead';
-    this.debug = debugModule('cloud-mirror:queue:' + this.queueName);
+    this.qid = slugid.nice();
   }
 
   /**
    * Declare required queue and dead letter queue, set up consumers
    */
   async init () {
```

Comment:
I *think* all of the init actions are actually sync and could be moved into the constructor... maybe I'll check into this and submit an extra commit in this PR